api/v1alpha1/fenceagentsremediationtemplate_webhook.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"io/fs"
+	"os"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/medik8s/fence-agents-remediation/pkg/validation"
+)
+
+var (
+	// webhookTemplateLog is for logging in this package.
+	webhookFARTemplateLog = logf.Log.WithName("fenceagentsremediationtemplate-resource")
+	// verify agent existence with os.Stat function
+	agentValidator = validation.NewAgentValidator(func(agent string) (fs.FileInfo, error) { return os.Stat(agent) })
```

Comment:
- since FAR is partof FARTemplate, and we want to validate FAR's agend, I would prefer to put actual validation and everything around into the FAR webhook
- since validation moved into its own package (which is ok), the webhook should not care about how validation happens. What about having a default validator which uses the filesystem in the validation package?